elastalert/test_rule.py

```diff
@@ -185,26 +183,23 @@ def mock_elastalert(self, elastalert):
         elastalert.get_hits = self.mock_hits
         elastalert.elasticsearch_client = mock.Mock()
 
-    def run_elastalert(self, rule, conf, args):
+    def run_elastalert(self, conf, args):
         """ Creates an ElastAlert instance and run's over for a specific rule using either real or mock data. """
 
-        # Load and instantiate rule
-        load_modules(rule)
-        conf['rules'] = [rule]
-
-        # If using mock data, make sure it's sorted and find appropriate time range
-        timestamp_field = rule.get('timestamp_field', '@timestamp')
         if args.json:
```

Comment:
Good point - it wouldn't work for this case. I can try and fix it. Problem is that there aren't any tests in this part of the world so I'm wary of doing too much major surgery.